src/connection/conn_string.rs

```diff
@@ -0,0 +1,432 @@
+//! Connection string parser
+
+use crate::*;
+use regex::Regex;
+use std::str::FromStr;
+
+pub struct ConnStringSettings {
+    pub user: Option<String>,
+    pub pass: Option<String>,
+    pub host: Option<String>,
+    pub port: Option<u16>,
+    pub db_name: String,
+    pub charset: Option<Charset>,
+    pub dialect: Option<Dialect>,
+    pub lib_path: Option<String>,
+    pub stmt_cache_size: Option<usize>,
+}
+
+/// Parse the connection string.
+///
+/// Basic string sintax: `firebird://{user}:{pass}@{host}:{port}/{db_name}?{options}`
+pub fn parse<S: Into<String>>(conn_s: S) -> Result<ConnStringSettings, FbError> {
+    let sconn = conn_s.into();
```

Comment:
There is no need to own the string.

```suggestion
pub fn parse(sconn: &str) -> Result<ConnStringSettings, FbError> {
```